Application/Contracts/Services/IStudentService.cs

```diff
@@ -0,0 +1,16 @@
+﻿
+
+using Application.Dtos;
+
+namespace Application.Contracts.IStudentService
+{
+    public interface IStudentService
+    {
+        Task<BaseResponse> RegisterStudent(CreateStudentRequest request);
+        Task<GetStudentResponse> GetStudentByEMail(string email);
+        Task<GetStudentResponse> GetStudentByStudentNumber(string studentNo);
+        Task<GetAllStudentResponse> GetAllStudents();
```

Comment:
Instead of GetAllStudents() simply use GetStudents(). Furthermore, GetStudents should be paginated with sorting options.